crates/ef-testing/src/evm_sequencer/sequencer/mod.rs

```diff
@@ -155,12 +155,11 @@ impl KakarotSequencer {
     }
 
     pub fn compute_starknet_address(&self, evm_address: &Address) -> StateResult<ContractAddress> {
-        let kakarot_address = *self.environment.kakarot_address.0.key();
         let base_class_hash = self.environment.base_account_class_hash.0;
 
         let constructor_args = {
             let evm_address: FeltSequencer = (*evm_address).try_into().unwrap(); // infallible
-            vec![kakarot_address, evm_address.into()]
+            vec![Felt::ONE, evm_address.into()]
```

Comment:
what is this Felt::ONE ?

Reply:
Good catch. Did not put more thoughts but should have.
It is the first argument of the constructor https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/kakarot/account.cairo#L510
Will look into it but seems not used ?

Reply:
it has to do with what `evm_address` gets serialized to (`[1, evm_address]`) https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/kakarot/accounts/uninitialized_account.cairo#L36-L41

Reply:
It seems there is something weird: 
1. `get_constructor_arguments` send back `(2, [1, evm_address)])` [here](https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/kakarot/account.cairo#L512) with `calldata_len = 2` and `calldata =  [1, evm_address)]`
2. It will be passed to the deploy_syscall [here](https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/backend/starknet.cairo#L81)
3. Hence the constructor of the uninitialized account will be called with the above tuple [here](https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/kakarot/accounts/uninitialized_account.cairo#L29)
4. But the initialize has only one parameter [here](https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/kakarot/accounts/account_contract.cairo#L46)

Hence it resolves to this:
```
library_call(
        class_hash=class_hash,
        function_selector=INITIALIZE_SELECTOR,
        calldata_size=2,
        calldata=[1, evm_address],
    );
```
Not sure how this could works as the function is 
```
func initialize{
        syscall_ptr: felt*,
        pedersen_ptr: HashBuiltin*,
        range_check_ptr,
        bitwise_ptr: BitwiseBuiltin*,
    }(evm_address) {
    ....
    }
```

Reply:
> Hence the constructor of the uninitialized account will be called with the above tuple [here](https://github.com/kkrt-labs/kakarot/blob/08e0330dc942b1af1c22340dd56727ef6a141e88/src/kakarot/accounts/uninitialized_account.cairo#L29) with calldata_len = 2 and calldata = [1, evm_address)]

Yes, this is true, the serialized data passed to the constructor is `[2, [1, evm_address]`. However this data is being deserialized into `calldata_len=1, calldata=[evm_address]`. Thus your library call is done with:

```
library_call(
        class_hash=class_hash,
        function_selector=INITIALIZE_SELECTOR,
        calldata_size=calldata_len(=1),
        calldata=[evm_address],
    );
 ```

 It makes more sense if I rename the input parameters of the constructor to `foo_len: felt, foo: felt*`
 => 
  ```
 @constructor
func constructor{syscall_ptr: felt*, pedersen_ptr: HashBuiltin*, range_check_ptr}(
    foo_len: felt, foo: felt*
) {
    library_call(
        class_hash=class_hash,
        function_selector=INITIALIZE_SELECTOR,
        calldata_size=foo_len,
        calldata=foo,
    );

    return ();
}
```

`constructor_calldata_len = 2, constructor_calldata=[1, evm_address]` are deserialized into `foo_len=1, foo=[evm_address]`

Reply:
Thank you for the detailed explanation. Understood now, I was missing context in `library_call` indeed.